### PR TITLE
GitLab error reporting - further features

### PR DESCRIPTION
**Pull request by selten**
Allow adding labels to GitLab issues. Allow user to define behaviour on opening a new issue when the previous one has been closed.

### CODE REVIEW

File: lib/galaxy/tools/error_reports/plugins/gitlab.py [hunk 1]
```diff
@@ -199,6 +204,26 @@ def submit_report(self, dataset, job, tool, **kwargs):
             log.error("GitLab error reporting - No connection to GitLab. Cannot report error to GitLab.")
             return ('Internal Error.', 'danger')
 
+    def _open_issue(self, error_message, error_title, gitlab_projecturl, gl_project, gl_userid, issue_cache_key):
+        """ Open an issue """
+        import gitlab
```
Comment (mvdbeek) [inline lib/galaxy/tools/error_reports/plugins/gitlab.py]: Can you pull the gitlab imports to the top and guard them like you do on line 48 ?

Comment (selten) [reply lib/galaxy/tools/error_reports/plugins/gitlab.py]: I believe I have done so now

File: lib/galaxy/tools/error_reports/plugins/gitlab.py [hunk 2]
```diff
@@ -13,6 +13,11 @@
     import urllib.parse as urllib
     urlparse = urllib
 
+try:
+    import gitlab
+except ImportError:
+    pass
```
Comment (mvdbeek) [inline lib/galaxy/tools/error_reports/plugins/gitlab.py]:
```suggestion
    gitlab = None
```

File: lib/galaxy/tools/error_reports/plugins/gitlab.py [hunk 3]
```diff
@@ -41,7 +46,8 @@ def __init__(self, **kwargs):
         self.gitlab_labels = kwargs.get("gitlab_labels", [])
 
         try:
-            import gitlab
+            if 'gitlab' not in sys.modules:
+                raise ImportError
```
Comment (mvdbeek) [inline lib/galaxy/tools/error_reports/plugins/gitlab.py]:
```suggestion
            if 'gitlab' is None:
                raise Exception("gitlab error reporting plugin is configured, but gitlab is not installed. Please install python-gitlab.")
```
Or something like this.

**Review by mvdbeek — APPROVED** [review]

File: lib/galaxy/tools/error_reports/plugins/gitlab.py [hunk 4]
```diff
@@ -13,6 +13,11 @@
     import urllib.parse as urllib
     urlparse = urllib
 
+try:
+    import gitlab
+except ImportError:
+    gitlab = None
+
```
Comment (mvdbeek) [inline lib/galaxy/tools/error_reports/plugins/gitlab.py]:
```suggestion

```

**Comment by jmchilton** [comment]
Awesome - thanks so much! This is cool work, thanks for the continued improvements.

**Pull request merged**